basic/sftp/src/test/java/org/springframework/integration/samples/sftp/EmbeddedSftpServer.java

```diff
@@ -78,35 +74,17 @@ public void afterPropertiesSet() throws Exception {
 		server.setFileSystemFactory(new VirtualFileSystemFactory(Paths.get(pathname)));
 	}
 
-	private PublicKey decodePublicKey() throws Exception {
+	private PublickeyAuthenticator getPublickeyAuthenticator() throws Exception {
 		InputStream stream = new ClassPathResource("META-INF/keys/sftp_rsa.pub").getInputStream();
-		byte[] keyBytes = StreamUtils.copyToByteArray(stream);
-		// strip any newline chars
-		while (keyBytes[keyBytes.length - 1] == 0x0a || keyBytes[keyBytes.length - 1] == 0x0d) {
-			keyBytes = Arrays.copyOf(keyBytes, keyBytes.length - 1);
-		}
-		byte[] decodeBuffer = Base64Utils.decode(keyBytes);
-		ByteBuffer bb = ByteBuffer.wrap(decodeBuffer);
-		int len = bb.getInt();
-		byte[] type = new byte[len];
-		bb.get(type);
-		if ("ssh-rsa".equals(new String(type))) {
-			BigInteger e = decodeBigInt(bb);
-			BigInteger m = decodeBigInt(bb);
-			RSAPublicKeySpec spec = new RSAPublicKeySpec(m, e);
-			return KeyFactory.getInstance("RSA").generatePublic(spec);
-
-		}
-		else {
-			throw new IllegalArgumentException("Only supports RSA");
-		}
-	}
-
-	private BigInteger decodeBigInt(ByteBuffer bb) {
-		int len = bb.getInt();
-		byte[] bytes = new byte[len];
-		bb.get(bytes);
-		return new BigInteger(bytes);
+		List<AuthorizedKeyEntry> keys = AuthorizedKeyEntry.readAuthorizedKeys(stream, true);
+		return new KeySetPublickeyAuthenticator(null, keys.stream().map(keyEntry -> {
```

Comment:
Can't we use an `AuthorizedKeysAuthenticator` instead?
The `new ClassPathResource("META-INF/keys/sftp_rsa.pub")` can be resolved to the `File` and that one to `Path` for the respective ctor:
```
    public AuthorizedKeysAuthenticator(Path file) {
```
?

Reply:
Excellent suggestion @artembilan, addressed in 6ced88ec06263704c55f7bcb635f36eafe7672ef

This plays nicely into a key-based setup as mentioned in [Server side security setup](https://github.com/apache/mina-sshd/blob/b88724358cda5d1f37d8ed2dd5ec24327d3eb95a/docs/server-setup.md?plain=1#L82) :+1: